code/modules/power/cable.dm

```diff
@@ -472,6 +502,13 @@ GLOBAL_LIST_INIT(cable_coil_recipes, list(new/datum/stack_recipe("cable restrain
 
 
 /obj/item/stack/cable_coil/update_icon()
+	if(icon_state == "cable_bridge")
+		name = "multilayer cable hub"
+		return
+	if(icon_state == "cablerelay-broken-cable")
+		name = "multi z layer cable hub"
+		return
+

```

Comment:
This proc should be setting the icon, not using it as a measure to set up something else.
Could you change the logic to follow something closer to that?
An example would be for the `layer_result` choice to change an index, and for this index to change to the appropriate `icon_state` on `update_icon()`

Reply:
yes but it already used for this

Reply:
By a different proc, you mean?
Still, no reason not to improve the code.

Reply:
done, no more name set in Update_icon()